libyul/backends/evm/StackHelpers.h

```diff
@@ -314,6 +322,22 @@ class Shuffler
 		// If we still need more slots, produce a suitable one.
 		if (ops.sourceSize() < ops.targetSize())
 		{
+			// If we do not need the current top and pushing another slot would make a position we need to fix unreachable,
+			// pop the current top.
```

Comment:
This seems to be enough to fix the issue at hand - but I still need to verify that this can't cause any other oscillation.
And, tricky as it may be, we should probably extract these cases as unit tests...